tests/unit/s2n_early_data_io_test.c

```diff
@@ -430,6 +430,65 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(server_conn));
         }
 
+        /* Early data rejected due to HRR, but received anyway and ignored.  */
+        {
+            struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(client_conn);
+            struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(server_conn);
+
+            struct s2n_test_io_pair io_pair = { 0 };
+            EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+
+            /* We run this multiple times because if the early data is incorrectly processed as a normal,
+             * unencrypted record, it is usually just ignored anyway because the record type is unknown.
+             * An error will only occur if the encryption happens to produce a known record type as the
+             * last non-padding byte.
+             *
+             * The chance this test produces a false negative:
+             * Ignoring the more complicated case where we strip "padding" off the end of the encrypted record,
+             * we handle 4 record types (HANDSHAKE, APPLICATION_DATA, ALERT, and CHANGE_CIPHER_SPEC).
+             * (((256 - 4) / 256) ^ 300) = 0.009, < 1%
+             */
```

Comment:
Q:  Even if we chances are low, when we do hit that false negative our CI will still fail right - how to do we keep track of that flakiness? Also, does this fix decrease the current flakiness of the session resumption self_talk_test  to < 1%, have we tested that?

Reply:
This doesn't really contribute to "flakiness" the way you're thinking about it. It won't cause our tests to occasionally fail.

What this is saying is that there is a <1% chance that this test will pass when it should fail. When I ran it without my fix, it failed on every run. It's like reverse "flakiness".